File: src/main/java/io/realworld/api/FavoritesApi.java
package io.realworld.api;

import io.realworld.model.CreateArticle201Response;

import io.vertx.core.Future;

public interface FavoritesApi  {
    Future<ApiResponse<CreateArticle201Response>> createArticleFavorite(String slug);
    Future<ApiResponse<CreateArticle201Response>> deleteArticleFavorite(String slug);
}

File: src/main/kotlin/io/realworld/api/FavoritesApi.kt
package io.realworld.api

import io.realworld.model.response.ApiResponse
import io.realworld.model.response.CreateArticle201Response
import io.vertx.core.Future

interface FavoritesApi {
  fun createArticleFavorite(slug: String): Future<ApiResponse<CreateArticle201Response>>
  fun deleteArticleFavorite(slug: String): Future<ApiResponse<CreateArticle201Response>>
}
